node/bft/src/primary.rs

```diff
@@ -1485,7 +1491,17 @@ impl<N: Network> Primary<N> {
             // If we do not have the certificate, request it.
             if !self.storage.contains_certificate(*certificate_id) {
                 trace!("Primary - Found a new certificate ID for round {round} from '{peer_ip}'");
-                // TODO (howardwu): Limit the number of open requests we send to a peer.
+                let mut outstanding_requests = self.outstanding_certificate_requests.lock();
+                let entry = outstanding_requests.entry((peer_ip, round));
+                // Get the number of outstanding requests for the peer. Note: the map is cleared on each round.
+                let num_outstanding_requests = entry.and_modify(|value| *value += 1).or_insert(1);
```

Comment:
Is this correct? We increment before checking if `num_oustanding_requests` exceeds the limit.